ports/renesas-ra/Makefile

```diff
@@ -436,6 +426,20 @@ $(TOP)/lib/fsp/README.md:
 	$(ECHO) "fsp submodule not found, fetching it now..."
 	(cd $(TOP) && git submodule update --init lib/fsp)
 
+CURRENT_FSP_HASH = $(patsubst +%,%,$(shell git -C $(TOP) submodule status lib/fsp))
+CHECK_FSP_VER:
+ifneq ($(CURRENT_FSP_HASH),$(EXPECTED_FSP_HASH))
```

Comment:
I think this test will always pass because both these variables are computed by calling `git status`.

Regardless of that, doing git submodule updates like this is very unconventional.  No other port does this so this reneses-ra port should not do it either.  Instead the submodule should just be updated as it normally would be.

In summary, please just remove this section that deals with the hash of the fsp lib.

Reply:
The `EXPECTED_FSP_HASH` can be defined per-board in `mpconfigboard.mk` files to override the default FSP version (the submodule tag/commit). The problem is the `e2studio` projects generated for every board, depend on the FSP version that was used to generate them. For example support for the new series was introduced in a more recent FSP, but all of the other boards use an older FSP, and our board `fsp-4.0.0`. Ideally FSP should be updated to the latest version and all the projects updated to match it, but that will be a lot of work, or it might be possible to remove the dependency on `e2studio` project files, I did notice that most of the generated files are not really used (only the vector table seem important, and the headers that enable the different stacks).

Reply:
We cannot have different boards requiring a different commit hash of the fsp submodule.

The fsp really needs to be updated to support the new board.  For example this is how stm32lib works when a new ST MCU is supported.

But, does the VK-RA6M5 here need a new fsp?

Reply:
> But, does the VK-RA6M5 here need a new fsp?

It does yes, it seems support for the series was introduced in `2.4.0` and then following releases added important fixes, see the releases below... For Portenta-C33 I based the project on `4.0.0`, I'm okay with regenerating the project, it's still local, but to update the the FSP to any newer version someone has to regenerate all the projects for all of the other boards.

https://github.com/renesas/fsp/releases/tag/v2.4.0

Reply:
> but to update the the FSP to any newer version someone has to regenerate all the projects for all of the other boards.

I don't understand why that's necessary... what files need to be regenerated when the fsp is updated?

Reply:
> I don't understand why that's necessary... what files need to be regenerated when the fsp is updated?

All the files in `boards/*/ra_gen`, `boards/*/ra_cfg`, `boards/*/src`...I'm not sure which files are actually used, to me it seems only the generated vector table and `ra_cfg/*` are important.

Reply:
Can you please explain how they are generated?  Can they be generated as part of the build process?

Reply:
Yes, your explanation is correct.  They are generated by e2 studio acoording to the project configuration.  So they can not be generated as part of the MicroPython build process. 
However, basically the FSP will try to keep conpatibility and will keep updating to support new devices.  The latest specific version of FSP might be easy to support renesas-ra port itself.    So if you all could agree to update FSP to the latest one or recent one, I agree with your decision and I am happy to check all EK and Clicker and will fix if there were any issues.

Reply:
OK, in that case, can we try to update the FSP to the latest version?  That task should be done as a separate, independent task with its own PR.

Reply:
I will try to update only the FSP to the latest version, now v4.4.0.   I think I can achieve this by `git submodule update --remote lib/fsp`.   Is this operation appropriate?

Reply:
This is untested, but it should be something like:

```bash
cd lib/fsp
git fetch origin
git checkout v4.4.0
cd ../
git add fsp
git commit -m "lib/fsp: update fsp to version 4.4.0."
```

Any link to documentation on how to update the fsp pack in e2 studio ?

Reply:
Thanks @iabdalkader san.  After updating, I tried to build but many compile errors happen.  Current version is very old, so it looks need to update ra_gen/* and others.  It might be not easy than I guess unfortunattely. 

>Any link to documentation on how to update the fsp pack in e2 studio ?

I hope the following FAQ link helps: 
https://en-support.renesas.com/knowledgeBase/20612472?_ga=2.71050256.1205374166.1682471818-1570124239.1633348365&_gac=1.183302740.1679888466.Cj0KCQajw2v-gBhC1ARIsAOQdKY3wL4ZrIc__3Btgo0mxi8wc2_1ci1Qqmyl6HhqpM1Rh_mflR35li1IaAn2vEALw_wcB&_gl=1*l9lvmi*_ga*MTU3MDEyNDIzOS4xNjMzMzQ4MzY1*_ga_D1706WVDQV*MTY4MjU5NDkwOC42ODEuMS4xNjgyNTk2MDYxLjAuMC4w

Reply:
Thank you that's exactly what I was looking for.

I agree it is a lot of work to update all of the projects, perhaps at some point in the future dependency on e2 studio generated files can be removed, but I'm okay with the current solution so it's up to @dpgeorge now.

Reply:
> but I'm okay with the current solution

Sorry I don't quite follow.  What is the "current solution" that you refer to?

Reply:
> Sorry I don't quite follow. What is the "current solution" that you refer to?

The current solution, which is in this PR, is to switch the `lib/fsp` version on the fly.

Reply:
I agree not having a dependency on e2 studio generated files at all would be great, but I don't know enough to say for sure if it's not needed, to me it seems that only the vector tables and the headers that enable/config different stacks are really needed, not sure if the other generated data structures are used.